detectron2/data/common.py

```diff
@@ -106,56 +110,101 @@ def __getitem__(self, idx):
                 )
 
 
+class NumpySerializedList(object):
+    """
+    A list-like object whose items are serialized and stored in a Numpy Array. When
+    passing this object to loader workers, they can use shared RAM from master process
+    instead of making a copy.
```

Comment:
I think "when passing this object to loader workers" is not clear: if you pass it as a `mp.Processs` argument or through a `mp.Queue`, they will not share memory. Change it to something like:

```
When forking a process that has NumpySerializedList, subprocesses can read the same list without 
triggering copy-on-write, therefore they will share RAM for the list. This avoids the issue in https://github.com/pytorch/pytorch/issues/13246
```

Reply:
Isn't the dataset passed as a `mp.Process` argument (https://github.com/pytorch/pytorch/blob/d1ba4c3a6d7a5007665419c57988b06d5b87e96e/torch/utils/data/dataloader.py#L1024)? When I benchmarked the newly added shared memory, I found that now even the `numpy` is not shared by subprocess, but I remembered that it was shared when we first added this.

Reply:
Yeah my bad. Passing it as a mp.Process argument should work (it's also just a fork). But through a mp.Queue would not work.

Reply:
Commented about numpy benchmark at https://github.com/facebookresearch/d2go/pull/405#issuecomment-1301510320